chat-commands.js

```diff
@@ -723,16 +723,33 @@ exports.commands = {
 
 	deletechatroom: 'deleteroom',
 	deletegroupchat: 'deleteroom',
-	deleteroom: function (target, room, user) {
+	deleteroom: function (target, room, user, connection, cmd) {
 		let roomid = target.trim();
-		if (!roomid) return this.parse('/help deleteroom');
-		let targetRoom = Rooms.search(roomid);
-		if (!targetRoom) return this.errorReply("The room '" + target + "' doesn't exist.");
-		if (room.isPersonal) {
+		if (!roomid) {
+			// allow deleting personal rooms without typing out the room name
+			if (room.isPersonal && cmd === "deletegroupchat") {
+				roomid = room.id;
+			} else {
+				return this.parse('/help deleteroom');
+			}
+		}
+
+		// find the room to be deleted, if it isn't the same the command is used in
```

Comment:
:| Disagree with this

Reply:
The reason the room name is passed is as a confirmation; it needs to equal the room the command is used in.